### PR TITLE
Grafana App Platform - velero devenv

### PR DESCRIPTION
**Pull request by charandas**
**What is this feature?**

Adds a devenv with Kind, Velero (utilizing minio for Object Storage). Some changes here in the authorizer will need thorough review. I added a few other K8s groups to utilize the current UserId/OrgId impersonation we do for the admin user.

**Why do we need this feature?**

To understand resource access patterns by standard K8s tools such as Velero.

**Who is this feature for?**

Grafana App Platform

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: Tiltfile
Comment (ryantxu) [inline Tiltfile:1]: this should live in the devenv/velero folder

Comment (charandas) [reply Tiltfile:1]: I thought its a matter of aesthetic but don't most people like to work from the root directory of the project? I couldn't find a way of remaining at root and invoking Tilt with a Tiltfile not at the root. Thoughts @toddtreece?

Comment (toddtreece) [reply Tiltfile:1]: two options i can think of at the moment:
* `cd devenv/velero && tilt up` seems fine to me
* add `tilt` to `.bingo` and add a `devenv-velero` make target that handles the `cd devenv/velero && tilt up`

File: pkg/infra/appcontext/user.go [hunk 1]
```diff
@@ -48,6 +53,14 @@ func User(ctx context.Context) (*user.SignedInUser, error) {
 	if ok {
 		for _, group := range k8sUserInfo.GetGroups() {
 			switch group {
+			case k8suser.AllAuthenticated:
```
Comment (ryantxu) [inline pkg/infra/appcontext/user.go:56]: AllAuthenticated??  will need to investigate this a bit more 😏

Comment (charandas) [reply pkg/infra/appcontext/user.go:56]: Usually, this group (`system:authenticated`) gets added as a supplemental group to `system:masters` and `system:apiserver`. With the User (Note: _not group_) `system:aggregator`, I saw that the only group assigned was `system:authenticated`. Once added, the discovery logs stopped reporting failures in parent kube-apiserver.

File: pkg/infra/appcontext/user.go [hunk 2]
```diff
@@ -48,6 +53,14 @@ func User(ctx context.Context) (*user.SignedInUser, error) {
...
+				// Doing this for system:aggregator User Name, which doesn't have a group
+				// other than system:authenticated and is in use for openapi/discovery by root apiserver
+				fallthrough
+			case VeleroServiceAccountGroup:
+				fallthrough
+			case KubeadmProvisionedAdminGroup:
```
Comment (charandas) [inline pkg/infra/appcontext/user.go:62]: Needed to add this, as for kind 1.29 (and other kubeadm based provisioned clusters also), this group gets made for the K8s admin/superuser.

Previously, <= 1.27, `system:masters` used to be enough to make Velero work.

File: pkg/services/grafana-apiserver/service.go [hunk 3]
```diff
@@ -233,9 +233,15 @@ func (s *service) start(ctx context.Context) error {
 	o := options.NewRecommendedOptions("/registry/grafana.app", Codecs.LegacyCodec(groupVersions...))
 	o.SecureServing.BindAddress = s.config.ip
 	o.SecureServing.BindPort = s.config.port
+
 	o.Authentication.RemoteKubeConfigFileOptional = true
 	o.Authorization.RemoteKubeConfigFileOptional = true
 
+	if s.config.aggregationEnabled {
+		o.Authentication.SkipInClusterLookup = true
+		o.Authentication.RequestHeader.ClientCAFile = s.config.requestHeaderClientCAFile
```
Comment (ryantxu) [inline pkg/services/grafana-apiserver/service.go:242]: is it possible to have aggregationEnabled without a requestHeaderClientCAFile?

do we ever want/need SkipInClusterLookup = false? (IIUC, we are never running in a state where we have config maps https://github.com/kubernetes/apiserver/blob/master/pkg/server/options/authentication.go#L206)

Comment (charandas) [reply pkg/services/grafana-apiserver/service.go:242]: `requestHeaderClientCAFile` is necessary with `SkipInClusterLookup`. Without the skip, the leaf server will try to look up `extension-apiserver-authentication` configmap which will require a kubeconfig to the parent server.

Here, we are skipping the need to provide that kubeconfig by providing a static CA that Grafana can trust.

**Pull request closed without merging**